commit 1746a06949656126037cd3ceb6678358f75cec07 (refs/changes/41/941/2)
Author: Joshua M. Clulow <jmc@joyent.com>
Date:   2016-11-22T18:26:05-08:00 (2 years, 11 months ago)
    
    joyent/manta-thoth#143 handle unexpected errors when fetching configuration
    Reviewed by: Bryan Cantrill <bryan@joyent.com>
    Approved by: Bryan Cantrill <bryan@joyent.com>

diff --git a/bin/thoth b/bin/thoth
index e63e21a..e9bd6c4 100755
--- a/bin/thoth
+++ b/bin/thoth
@@ -3499,8 +3499,15 @@ var configure = function (cb)
 	client = connect();
 
 	client.get(object, function (err, stream, res) {
-		if (err && err.code == 'ResourceNotFound')
-			return (cb(client, 'from source'));
+		if (err) {
+			if (err.code === 'ResourceNotFound') {
+				cb(client, 'from source');
+				return;
+			}
+
+			fatal('could not load thoth configuration (' +
+			    object + ') from Manta: ' + err);
+		}
 
 		var output = '';
 
diff --git a/package.json b/package.json
index 6c0e252..5abd9a2 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
 	"name": "manta-thoth",
 	"description": "Manta-based system for core and crash dump analysis",
-	"version": "0.2.0",
+	"version": "0.2.1",
 	"author": "Joyent (joyent.com)",
 	"dependencies": {
 		"manta": "1.6.0",
